### PR TITLE
Use xcrun to find Mac SDKs instead of searching directories

### PR DESCRIPTION
**Pull request by dschuff**
Also: 
* Use the SDKROOT environment variable rather than the -isysroot flag to configure clang.
* Add a manual include of the emscripten libcxx headers since those are not included in the SDK.

### CODE REVIEW

**Comment by dschuff** [comment]
Most of the benchmarks work now, although some of the library-builder-based ones still do not.

File: tests/test_benchmark.py [hunk 1]
```diff
@@ -121,7 +121,10 @@ def __init__(self, name, cc, cxx, args=[OPTIMIZATIONS]):
   def build(self, parent, filename, args, shared_args, emcc_args, native_args, native_exec, lib_builder, has_output_parser):
     self.parent = parent
     if lib_builder:
-      native_args += lib_builder(self.name, native=True, env_init={'CC': self.cc, 'CXX': self.cxx, 'CXXFLAGS': "-Wno-c++11-narrowing"})
+      env = {'CC': self.cc, 'CXX': self.cxx, 'CXXFLAGS': "-Wno-c++11-narrowing"}
+      env.update(shared.get_clang_native_env())
+      print(env)
```
Comment (kripken) [inline tests/test_benchmark.py]: remove debug print?

**Comment by kripken** [comment]
I don't know MacOS that well though, so take my lgtm with a grain of salt...

File: tests/test_benchmark.py (hunk 1 above)
Comment (dschuff) [reply tests/test_benchmark.py]: Done.

**Comment by dschuff** [comment]
Yeah, I only sort-of know macos things. As I mentioned not everything works yet, and I'm not sure I'll go the rest of the way to finish unless and until we want to start running benchmarks on mac.

File: tools/shared.py [hunk 2]
```diff
@@ -669,7 +653,11 @@ def get_clang_native_env():
     return CACHED_CLANG_NATIVE_ENV
   env = os.environ.copy()
 
-  if WINDOWS:
+  if MACOS:
+    path = run_process(['xcrun', '--show-sdk-path'], stdout=PIPE).stdout.strip()
+    logging.debug('Using MacOS SDKROOT: ' + path)
+    env['SDKROOT'] = path
```
Comment (sbc100) [inline tools/shared.py]: Is this a macOS specific thing?

Comment (dschuff) [reply tools/shared.py]: Yes, `$SDKROOT` is used by clang and XCode tools to find the MacOS/iOS SDKs on the system, and the `xcrun` and `xcode-select` tools are command-line utilities for managing them.

**Pull request merged**